modin/utils.py

```diff
@@ -535,3 +535,86 @@ def instancer(_class):
         Instance of `_class`.
     """
     return _class()
+
+
+def value_or_default(value, default, allow_calling_default=True, none_value=None):
+    """
+    Return `default` if the `value` is `none_value` or the `value` otherwise.
+
+    Parameters
+    ----------
+    value : object
+        Value for the none check.
+    default : object or type
+        Default value to return if `value` is none. If `default` is a type and
+        `allow_calling_default` is True return an instance of the `default` type.
+    allow_calling_default : bool, default: True
+        Whether to return ``default()`` if `default` is a type.
+    none_value : object, default: None
+        Reference for the None value.
+
+    Returns
+    -------
+    object
+    """
+    return (
+        value
+        if value is not none_value
+        else (
+            default()
+            if isinstance(default, type) and allow_calling_default
+            else default
+        )
+    )
+
+
+def partial_method_late_binding(func, *args, **kwargs):
```

Comment:
`functools.partialmethod` does not provide the required result
<details><summary>example</summary>

```python
>>> class Base:
...     def general_method(self, x):
...         print(f"Base: {x}")
...     specific_method_functools = functools.partialmethod(general_method, x=1)
...     specific_method_modinutils = partial_method_late_binding(general_method, x=1)
...
>>> class Derived(Base):
...     def general_method(self, x):
...         print(f"Derived: {x}")
...
>>> # Functools:
>>> Derived().specific_method_functools()
Base: 1
>>> # Desired:
>>> Derived().specific_method_modinutils()
Derived: 1
```

</details>